howso/feature_residuals.amlg

```diff
@@ -621,6 +625,121 @@
 						(unzip feature_residuals_lists numeric_continuous_indices)
 					)
 				)
+			;smape = |predicted - actual| / ((|actual| + |predicted|) / 2) * 100
+			smape_map
+				(zip
+					numeric_continuous_features
+					;iterate over continuous features, where (current_value) is a list of react tuples for each feature
+					(map
+						(lambda
+							(if (size (current_value))
+								(*
+									(/
+										;sum of the absolute percent errors
+										(apply "+"
+											(map
+												(lambda
+													(let
+														(assoc
+															smape_numerator
+																(abs (-
+																	(get (current_value 1) predicted_offset)
+																	(get (current_value 1) actual_offset)
+																))
+															smape_denominator
+																(/ (+
+																		(abs (get (current_value 1) actual_offset))
+																		(abs (get (current_value 1) predicted_offset))
+																	)
+																	2
+																)
+														)
+														(if (= smape_denominator 0)
+															;if the denominator is 0, that means actual and predictd are both 0
+															; which should have a error of 0
+															0
+															(/ smape_numerator smape_denominator)
+														)
+													)
+												)
+												(current_value)
+											)
+										)
+										(size (current_value))
+									)
+									100
+								)
+							)
+						)
+						(unzip feature_residuals_lists numeric_continuous_indices)
+					)
+				)
+			;adjusted_smape = |predicted - actual| / ((|actual| + |predicted|) / 2) * 100 where min gap / 2 is added to both the actual and predicted values.
+			adjusted_smape_map
+				(zip
+					numeric_continuous_features
+					;iterate over continuous features, where (current_value) is a list of react tuples for each feature
+					(map
+						(lambda
+							(if (size (current_value))
+								(let (assoc
+									adjusted_min_gap
+										(/
+											(if (contains_index !editDistanceFeatureTypesMap (get numeric_continuous_features (current_index 1)) )
```

Comment:
can a "numeric_continuous" feature ever be an editDistance feature?

Reply:
actually don't think so